### ISSUE TITLE
Android: app crash RNCWebViewManager$RNCWebView, size 1080x176205 exceeds max size 8192

### ISSUE THREAD

**Issue by phuongwd** [react-native-community/react-native-webview#906]
**Bug description:**

crash log:

> JNI DETECTED ERROR IN APPLICATION: JNI CallObjectMethod called with pending exception java.lang.IllegalStateException: Unable to create layer for RNCWebViewManager$RNCWebView, size 1080x176205 exceeds max size 8192

<details>
  <summary>Click to open full crash log</summary>
  
 ```
2019-09-26 17:09:35.469 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504] JNI DETECTED ERROR IN APPLICATION: JNI CallObjectMethod called with pending exception java.lang.IllegalStateException: Unable to create layer for RNCWebViewManager$RNCWebView, size 1080x176205 exceeds max size 8192
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   at void android.os.MessageQueue.nativePollOnce(long, int) (MessageQueue.java:-2)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   at android.os.Message android.os.MessageQueue.next() (MessageQueue.java:325)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   at void android.os.Looper.loop() (Looper.java:142)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6938)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:327)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:1374)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504] 
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]     in call to CallObjectMethod
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]     from void android.os.MessageQueue.nativePollOnce(long, int)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504] "main" prio=5 tid=1 Runnable
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   | group="main" sCount=0 dsCount=0 flags=0 obj=0x73cc9f50 self=0x75870c0a00
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   | sysTid=5701 nice=-4 cgrp=default sched=0/0 handle=0x758c74a9c8
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   | state=R schedstat=( 12974759229 1569599587 12905 ) utm=1012 stm=285 core=4 HZ=100
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   | stack=0x7fc3429000-0x7fc342b000 stackSize=8MB
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   | held mutexes= "mutator lock"(shared held)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   native: #00 pc 00000000003aeb84  /system/lib64/libart.so (_ZN3art15DumpNativeStackERNSt3__113basic_ostreamIcNS0_11char_traitsIcEEEEiP12BacktraceMapPKcPNS_9ArtMethodEPv+212)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   native: #01 pc 000000000047deac  /system/lib64/libart.so (_ZNK3art6Thread9DumpStackERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEEbP12BacktraceMapb+348)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   native: #02 pc 00000000002e6280  /system/lib64/libart.so (_ZN3art9JavaVMExt8JniAbortEPKcS2_+1040)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   native: #03 pc 00000000002e6724  /system/lib64/libart.so (_ZN3art9JavaVMExt9JniAbortVEPKcS2_St9__va_list+116)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   native: #04 pc 000000000010b5dc  /system/lib64/libart.so (_ZN3art11ScopedCheck6AbortFEPKcz+172)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   native: #05 pc 000000000010b068  /system/lib64/libart.so (_ZN3art11ScopedCheck11CheckThreadEP7_JNIEnv+536)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   native: #06 pc 000000000010937c  /system/lib64/libart.so (_ZN3art11ScopedCheck5CheckERNS_18ScopedObjectAccessEbPKcPNS_12JniValueTypeE+860)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   native: #07 pc 000000000010eb84  /system/lib64/libart.so (_ZN3art8CheckJNI13CheckCallArgsERNS_18ScopedObjectAccessERNS_11ScopedCheckEP7_JNIEnvP8_jobjectP7_jclassP10_jmethodIDNS_10InvokeTypeEPKNS_7VarArgsE+132)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   native: #08 pc 000000000010d9f8  /system/lib64/libart.so (_ZN3art8CheckJNI11CallMethodVEPKcP7_JNIEnvP8_jobjectP7_jclassP10_jmethodIDSt9__va_listNS_9Primitive4TypeENS_10InvokeTypeE+680)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   native: #09 pc 00000000000fa948  /system/lib64/libart.so (_ZN3art8CheckJNI16CallObjectMethodEP7_JNIEnvP8_jobjectP10_jmethodIDz+184)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   native: #10 pc 0000000000136320  /system/lib64/libandroid_runtime.so (_ZN7android26NativeDisplayEventReceiver13dispatchVsyncElij+48)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   native: #11 pc 0000000000042bb4  /system/lib64/libandroidfw.so (_ZN7android22DisplayEventDispatcher11handleEventEiiPv+164)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   native: #12 pc 0000000000016854  /system/lib64/libutils.so (_ZN7android6Looper9pollInnerEi+836)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   native: #13 pc 0000000000016484  /system/lib64/libutils.so (_ZN7android6Looper8pollOnceEiPiS1_PPv+116)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   native: #14 pc 000000000016057c  /system/lib64/libandroid_runtime.so (???)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   native: #15 pc 00000000001effdc  /system/framework/arm64/boot-framework.oat (Java_android_os_MessageQueue_nativePollOnce__JI+140)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   at android.os.MessageQueue.nativePollOnce(Native method)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   at android.os.MessageQueue.next(MessageQueue.java:325)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   at android.os.Looper.loop(Looper.java:142)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   at android.app.ActivityThread.main(ActivityThread.java:6938)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   at java.lang.reflect.Method.invoke(Native method)
2019-09-26 17:09:35.470 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504]   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: java_vm_ext.cc:504] 
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:516] Runtime aborting...
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:516] 
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524] JNI DETECTED ERROR IN APPLICATION: JNI CallObjectMethod called with pending exception java.lang.IllegalStateException: Unable to create layer for RNCWebViewManager$RNCWebView, size 1080x176205 exceeds max size 8192
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   at void android.os.MessageQueue.nativePollOnce(long, int) (MessageQueue.java:-2)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   at android.os.Message android.os.MessageQueue.next() (MessageQueue.java:325)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   at void android.os.Looper.loop() (Looper.java:142)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6938)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   at java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) (Method.java:-2)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   at void com.android.internal.os.Zygote$MethodAndArgsCaller.run() (Zygote.java:327)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:1374)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524] 
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]     in call to CallObjectMethod
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]     from void android.os.MessageQueue.nativePollOnce(long, int)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524] "main" prio=5 tid=1 Runnable
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   | group="main" sCount=0 dsCount=0 flags=0 obj=0x73cc9f50 self=0x75870c0a00
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   | sysTid=5701 nice=-4 cgrp=default sched=0/0 handle=0x758c74a9c8
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   | state=R schedstat=( 12974759229 1569599587 12905 ) utm=1012 stm=285 core=4 HZ=100
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   | stack=0x7fc3429000-0x7fc342b000 stackSize=8MB
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   | held mutexes= "mutator lock"(shared held)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   native: #00 pc 00000000003aeb84  /system/lib64/libart.so (_ZN3art15DumpNativeStackERNSt3__113basic_ostreamIcNS0_11char_traitsIcEEEEiP12BacktraceMapPKcPNS_9ArtMethodEPv+212)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   native: #01 pc 000000000047deac  /system/lib64/libart.so (_ZNK3art6Thread9DumpStackERNSt3__113basic_ostreamIcNS1_11char_traitsIcEEEEbP12BacktraceMapb+348)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   native: #02 pc 00000000002e6280  /system/lib64/libart.so (_ZN3art9JavaVMExt8JniAbortEPKcS2_+1040)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   native: #03 pc 00000000002e6724  /system/lib64/libart.so (_ZN3art9JavaVMExt9JniAbortVEPKcS2_St9__va_list+116)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   native: #04 pc 000000000010b5dc  /system/lib64/libart.so (_ZN3art11ScopedCheck6AbortFEPKcz+172)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   native: #05 pc 000000000010b068  /system/lib64/libart.so (_ZN3art11ScopedCheck11CheckThreadEP7_JNIEnv+536)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   native: #06 pc 000000000010937c  /system/lib64/libart.so (_ZN3art11ScopedCheck5CheckERNS_18ScopedObjectAccessEbPKcPNS_12JniValueTypeE+860)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   native: #07 pc 000000000010eb84  /system/lib64/libart.so (_ZN3art8CheckJNI13CheckCallArgsERNS_18ScopedObjectAccessERNS_11ScopedCheckEP7_JNIEnvP8_jobjectP7_jclassP10_jmethodIDNS_10InvokeTypeEPKNS_7VarArgsE+132)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   native: #08 pc 000000000010d9f8  /system/lib64/libart.so (_ZN3art8CheckJNI11CallMethodVEPKcP7_JNIEnvP8_jobjectP7_jclassP10_jmethodIDSt9__va_listNS_9Primitive4TypeENS_10InvokeTypeE+680)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   native: #09 pc 00000000000fa948  /system/lib64/libart.so (_ZN3art8CheckJNI16CallObjectMethodEP7_JNIEnvP8_jobjectP10_jmethodIDz+184)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   native: #10 pc 0000000000136320  /system/lib64/libandroid_runtime.so (_ZN7android26NativeDisplayEventReceiver13dispatchVsyncElij+48)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   native: #11 pc 0000000000042bb4  /system/lib64/libandroidfw.so (_ZN7android22DisplayEventDispatcher11handleEventEiiPv+164)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   native: #12 pc 0000000000016854  /system/lib64/libutils.so (_ZN7android6Looper9pollInnerEi+836)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   native: #13 pc 0000000000016484  /system/lib64/libutils.so (_ZN7android6Looper8pollOnceEiPiS1_PPv+116)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   native: #14 pc 000000000016057c  /system/lib64/libandroid_runtime.so (???)
2019-09-26 17:09:35.471 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   native: #15 pc 00000000001effdc  /system/framework/arm64/boot-framework.oat (Java_android_os_MessageQueue_nativePollOnce__JI+140)
2019-09-26 17:09:35.472 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   at android.os.MessageQueue.nativePollOnce(Native method)
2019-09-26 17:09:35.472 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   at android.os.MessageQueue.next(MessageQueue.java:325)
2019-09-26 17:09:35.472 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   at android.os.Looper.loop(Looper.java:142)
2019-09-26 17:09:35.472 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   at android.app.ActivityThread.main(ActivityThread.java:6938)
2019-09-26 17:09:35.472 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   at java.lang.reflect.Method.invoke(Native method)
2019-09-26 17:09:35.472 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
2019-09-26 17:09:35.472 5701-5701/com.demo.app A/zygote64: runtime.cc:524]   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
2019-09-26 17:09:35.472 5701-5701/com.demo.app A/zygote64: runtime.cc:524] 
    
    --------- beginning of crash
2019-09-26 17:09:35.473 5701-5701/com.demo.app A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 5701
```
</details>

**To Reproduce:**

- Large html
- Webview inside ScrollView

**Environment:**
 - OS: Android
 - OS version: 8.0.0
 - react-native version: 0.60.5
 - react-native-webview version: 7.2.4

**Comment by phuongwd** [comment]
**Update:**
this issue related to the pull request [fix(Android): hardware acceleration issue](https://github.com/react-native-community/react-native-webview/pull/854) (#854)
App does not crash if I revert it as below:
```
@ReactProp(name = "androidHardwareAccelerationDisabled")
  public void setHardwareAccelerationDisabled(WebView view, boolean disabled) {
    if (disabled) {
      view.setLayerType(View.LAYER_TYPE_SOFTWARE, null);
    } else {
      view.setLayerType(View.LAYER_TYPE_NONE, null);
    }
  }
```

**Comment by thanakij** [comment]
@phuongwd Could you help us verify if this is the case for WebView by:
(1) You have some <ScrollView> wrapping around <WebView>, can you help change to and see if the crash disappeared.
(2) Next, please get back the <ScrollView> and try to replace the (probably large) content from <WebView> with plain <Text> to see if the same crash still happen without WebView.

**Comment by esjs** [comment]
I've had similar  problem, what helped me is to wrap WebView inside
<View style={{flex: 1}}>
  <WebView ... />
</View>

Hope it would solve your problem as well @phuongwd

**Comment by hqdai** [comment]
I think it's related to RNN (React-native-navigation) animate, I got same issue and it has fixed with RNN 3.2.0 on React-Native 60.5.

**Comment by phuongwd** [comment]
@hqdai thank you for your verification.

@thanakij thanks 🥰
